src/main/java/com/kobylynskyi/graphql/codegen/mapper/GraphQLTypeMapper.java

```diff
@@ -312,6 +314,17 @@ default NamedDefinition getLanguageType(MappingContext mappingContext, String gr
             if (!Utils.isEmpty(annotationsToAdd)) {
                 annotations.addAll(annotationsToAdd);
             }
+        } else if (Boolean.TRUE.equals(mappingContext.getGenerateJacksonTypeIdResolver()) && isUnion) {
+            annotations.add("com.fasterxml.jackson.annotation.JsonTypeInfo(use = " +
+                    "com.fasterxml.jackson.annotation.JsonTypeInfo.Id.NAME, property = \"__typename\")");
+            String modelPackageName = DataModelMapper.getModelPackageName(mappingContext);
+            if (modelPackageName == null) {
+                modelPackageName = "";
+            } else if (Utils.isNotBlank(modelPackageName)) {
+                modelPackageName += ".";
+            }
+            annotations.add("com.fasterxml.jackson.databind.annotation.JsonTypeIdResolver(" + modelPackageName +
```

Comment:
`add annotation` , It's not universal logic. It can't be put here.
Need to put in `JavaGraphqlTypeMapper`


```
 default List<String>  getExtractAnnotations(MappingContext mappingContext, boolean isUnion) {
        return new ArrayList<>();
    }

annotations.addAll(getExtractAnnotations(mappingContext,isUnion));
```


In JavaGraphQLTypeMapper
```
    public List<String> getExtractAnnotations(MappingContext mappingContext, boolean isUnion) {
        List<String> defaults = new ArrayList<>();
        if (Boolean.TRUE.equals(mappingContext.getGenerateJacksonTypeIdResolver()) && isUnion) {
            defaults.add("com.fasterxml.jackson.annotation.JsonTypeInfo(use = " +
                    "com.fasterxml.jackson.annotation.JsonTypeInfo.Id.NAME, property = \"__typename\")");
            String modelPackageName = DataModelMapper.getModelPackageName(mappingContext);
            if (modelPackageName == null) {
                modelPackageName = "";
            } else if (Utils.isNotBlank(modelPackageName)) {
                modelPackageName += ".";
            }
            defaults.add("com.fasterxml.jackson.databind.annotation.JsonTypeIdResolver(" + modelPackageName +
                    "GraphqlJacksonTypeIdResolver.class)");
        }
        return defaults;
    }
```

In Kotlin
```
    public List<String> getExtractAnnotations(MappingContext mappingContext, boolean isUnion) {
        List<String> defaults = new ArrayList<>();
        if (Boolean.TRUE.equals(mappingContext.getGenerateJacksonTypeIdResolver()) && isUnion) {
            defaults.add("com.fasterxml.jackson.annotation.JsonTypeInfo(use = " +
                    "com.fasterxml.jackson.annotation.JsonTypeInfo.Id.NAME, property = \"__typename\")");
            String modelPackageName = DataModelMapper.getModelPackageName(mappingContext);
            if (modelPackageName == null) {
                modelPackageName = "";
            } else if (Utils.isNotBlank(modelPackageName)) {
                modelPackageName += ".";
            }
            defaults.add("com.fasterxml.jackson.databind.annotation.JsonTypeIdResolver(" + modelPackageName +
                    "GraphqlJacksonTypeIdResolver::class)");
        }
        return defaults;
    }
```

Reply:
I had a look to your PR and merged it. Thanks !

Reply:
A single test error was found and PR was submitted.